### PR TITLE
Fix issue 152

### LINKED ISSUE #152: Make sure that parent datasets are present in the same environment as all its children
The `metadata_identifier` element in the MMD-files have identifiers on the form `naming_authority:uuid`.

- [ ] The parent-child relationships defined in child datasets must be updated to the correct `naming_authority:uuid` of the parent

- [ ] Because of the way the discovery metadata databases are designed, both parent and child datasets should be in the same environment. Therefore we need to update the `naming_authority` parts of the id's pointing to parents, both for existing datasets and those coming into dmci. I.e., DMCI must update the `naming_authority` in the same way as it is doing for the `metadata_id`, by adding "-dev" or "-staging".

- [ ] Since the parents need to be present in the same environment, we also need to check that the referenced parent exists, and reject the MMD file if the referenced parent does not exist.

### PR DESCRIPTION
**Pull request by charlienegri**
**Summary**: fixes #152 

**Related issue**:

**Suggested reviewer(s)**:

**Reviewer checklist**:

* [x] The headers of all files contain a reference to the repository license
* [x] 100% test coverage of new code - meaning:
  * [x] The overall test coverage increased or remained the same as before
  * [x] Every function is accompanied with a test suite
  * [x] Tests are both positive (testing that the function work as intended with valid data) and negative (testing that the function behaves as expected with invalid data, e.g., that correct exceptions are thrown)
  * [x] Functions with optional arguments have separate tests for all options
* [x] Examples are supported by doctests
* [x] All tests are passing
* [x] All names (e.g., files, classes, functions, variables) are explicit
* [x] Documentation (as docstrings) is complete and understandable

The checklist is based on the S-ENDA conventions and definition of done (see [General Conventions](https://s-enda-documentation.readthedocs.io/en/latest/general_conventions.html)). The above points are not necessarily relevant to all contributions. In that case, please add a short explanation to help the reviewer.

### CODE REVIEW

**Comment by charlienegri** [comment]
managed 100% coverage but I broke compatibility with python 3.7 😅

**Comment by charlienegri** [comment]
added a return (+ test) that should fix #156

File: dmci/api/worker.py
Comment (johtoblan) [inline dmci/api/worker.py:102]: Why not use the r-formatting?

**Review by johtoblan — APPROVED** [review]
LGTM

File: dmci/api/worker.py
Comment (charlienegri) [reply dmci/api/worker.py:102]: I do not remember why I had changed this, r-formatting should be fine...

**Review by johtoblan — APPROVED** [review]

**Comment by charlienegri** [comment]
I added the check at line 125 for pure redundancy, because we said the fact that a parent does not have a correct identifier should not be possible once we fix py-mmd-tools, I will edit the return, you are right that it misses the 3rd argument
the test for line 85 was written to expect a return of 2 parameters as long as I can see:
```
    # Invalid data format
    passData = readFile(passFile)
    assert passWorker.validate(passData) == (False, "Input must be bytes type")
```
I will fix this too

**Comment by charlienegri** [comment]
@magnarem can you try again?

**Comment by magnarem** [comment]
Tested locally with a few test-mmd files. Works ok now.
Updated the dockstring.

**Review by magnarem — APPROVED** [review]
Reviewed and approved

**Review by magnarem — APPROVED** [review]
Approving this MR. But we should probably still wait until #242-py-mmd-tools have been merged.

**Pull request merged**
